src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -1255,25 +1271,15 @@ export default class RouteModel extends BaseModel {
     }
 
     _overwriteRouteBetweenWaypointNames(startWaypointName, endWaypointName, routeModel) {
-        const initialIndexOfDivergentLegModel = this._findIndexOfLegContainingWaypointName(startWaypointName);
-        const initialIndexOfConvergentLegModel = this._findIndexOfLegContainingWaypointName(endWaypointName);
-
-        // this must happen first; Array.splice() is mutating the #_legCollection
-        const endingLegCollection = this._legCollection.splice(initialIndexOfConvergentLegModel);
-
-        this._legCollection.splice(initialIndexOfDivergentLegModel + 1);
-
-        const beginningLegCollection = this._legCollection;
-
-        this._legCollection = [
-            ...beginningLegCollection,
-            ...endingLegCollection
-        ];
-
         const indexOfDivergentLegModel = this._findIndexOfLegContainingWaypointName(startWaypointName);
         const indexOfConvergentLegModel = this._findIndexOfLegContainingWaypointName(endWaypointName);
         const amendedDivergentLegModels = this._createAmendedDivergentLeg(indexOfDivergentLegModel, startWaypointName);
         const amendedConvergentLegModels = this._createAmendedConvergentLeg(indexOfConvergentLegModel, endWaypointName);
+        const endingLegCollection = this._legCollection.splice(indexOfConvergentLegModel + 1);
+
+        this._legCollection.splice(indexOfDivergentLegModel);
```

Comment:
I would suggest first creating a copy of the `_legCollection` at the top of this method:
```js
const legCollection = this._legCollection.slice(0);
```

Then you can mutate the `nextLegCollection` instead of directly mutating the `#_legCollection` instance.  This is a subtle change but it means that you only ever reassign `#_legCollection` with a finished update instead of part of the finished product.  Doing it this way makes the logic a little clearer too.

Reply:
along those same lines, there should be a method on the `LegCollection` (or `BaseCollection`) class that returns a copy.  This logic should _never_ be done inline with a collection class.

Reply:
first comment done, and for the second, the `_legCollection` is actually just a simple array currently, so can't do that at the moment. But added #925 to add a copy method to the base classes.

done